src/main/java/org/gradle/profiler/InstrumentingProfiler.java

```diff
@@ -0,0 +1,172 @@
+package org.gradle.profiler;
+
+import java.io.IOException;
+
+/**
+ * A profiler that can:
+ *
+ * <ul>
+ * <li>Instrument a JVM using JVM args</li>
+ * <li>Start recording on JVM start, using JVM args</li>
+ * <li>Capture snapshot on JVM exit, using JVM args</li>
+ * <li>Use some communication mechanism to start recording in an instrumented JVM that is currently running.</li>
+ * <li>Use some communication mechanism to pause recording in an instrumented JVM that is currently running.</li>
+ * <li>Use some communication mechanism to capture a snapshot from an instrumented JVM that is currently running.</li>
+ * </ul>
+ *
+ * <p>The profiler may support starting recording multiple times for a given JVM. The implementation should indicate this by overriding {@link #canRestartRecording()}.</p>
+ */
+public abstract class InstrumentingProfiler extends Profiler {
+    @Override
+    public JvmArgsCalculator newJvmArgsCalculator(ScenarioSettings settings) {
+        validate(settings);
+        if (!settings.getScenario().getInvoker().isReuseDaemon()) {
+            // When the daemon is not reused, there is no need to instrument the warm ups
```

Comment:
I'd very much prefer method-level comments instead of inline ones, as the latter grow stale so quickly.

Reply:
I've updated the javadocs.